clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -221,7 +265,7 @@ public boolean isReady(Node node, long now) {
      * @param node The node
      */
     private boolean canSendRequest(String node) {
-        return connectionStates.isConnected(node) && selector.isChannelReady(node) && inFlightRequests.canSendMore(node);
+        return this.requiredApiVersions == null ? connectionStates.isConnected(node) : connectionStates.isReady(node) && selector.isChannelReady(node) && inFlightRequests.canSendMore(node);
```

Comment:
I think it would be better for `READY` to be the state where one sends requests. This means that if the api version requests are not needed, we can automatically transition from `CONNECTED` to `READY`. Thoughts? cc @hachikuji.